app/Jobs/CloneImagesOrVideos.php

```diff
@@ -0,0 +1,480 @@
+<?php
+
+namespace Biigle\Jobs;
+
+use Biigle\Http\Requests\CloneVolume;
+use Biigle\Image;
+use Biigle\ImageAnnotation;
+use Biigle\ImageAnnotationLabel;
+use Biigle\ImageLabel;
+use Biigle\Project;
+use Biigle\Traits\ChecksMetadataStrings;
+use Biigle\Video;
+use Biigle\VideoAnnotation;
+use Biigle\VideoAnnotationLabel;
+use Biigle\VideoLabel;
+use Biigle\Volume;
+use \Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Storage;
+use Ramsey\Uuid\Uuid;
+
+class CloneImagesOrVideos extends Job implements ShouldQueue
+{
+    use InteractsWithQueue, SerializesModels, ChecksMetadataStrings;
+
+
+    /**
+     * The target project.
+     *
+     * @var Project
+     */
+    public Project $project;
+
+    /**
+     * The volume to clone.
+     *
+     * @var Volume
+     */
+    public Volume $volume;
+
+    /**
+     * The new name of the volume clone.
+     *
+     * @var string
+     **/
+    public string $cloneName;
+
+    /**
+     * Array containing file ids.
+     *
+     * @var array
+     **/
+    public array $onlyFiles;
+
+    /**
+     * Boolean for cloning annotation labels.
+     *
+     * @var bool
+     **/
+    public bool $cloneAnnotations;
+
+    /**
+     * Array containing annotation label ids.
+     *
+     * @var array
+     **/
+    public array $onlyAnnotationLabels;
+
+    /**
+     * Boolean for cloning file labels.
+     *
+     * @var bool
+     **/
+    public bool $cloneFileLabels;
+
+    /**
+     * Array containing file label ids.
+     *
+     * @var array
+     **/
+    public array $onlyFileLabels;
```

Comment:
Please also add this:

```php
/**
 * Ignore this job if the project or volume does not exist any more.
 *
 * @var bool
 */
protected $deleteWhenMissingModels = true;
```

The project and volume models will be serialized when the job is pushed to the queue. Later when the job is processed and one of the models doesn't exist any more (because it was deleted in the meantime), there would be an error if the variable above is not set to `true`. With the variable `true` the job will simply be deleted in that case.